### PR TITLE
Fix for update products via csv file (fix for 22028)

### PR DESCRIPTION
**Pull request by mtwegrzycki**
### Description (*)
It is fix for issue #22028

In query for selecting ids to indexing after import products, I changed "between" into "in",
Before change, when products were updated via csv file import in Admin panel, to indexing process were selected all products from range between the smallest and the biggest products id from csv file. When the ids were from wide range, it caused selecting a huge amount of data and attempt to inserting them to catalog_product_index_price_temp. If data amount was to big, it caused error
After change, for indexing process are selected only products from csv file.

Example:
Before change, when we try to update two products with id 30 and 31, only two products were selected to indexing. 
When products have ids 30 and 50, it caused, that 21 products were selected to indexing (All products with ids between 30 and 50.

After change, in both cases described above, only two products will be selected to indexing.

### Fixed Issues (if relevant)
No related issues found.

### Manual testing scenarios (*)
The same scenario can be applied to reproduce bug and check fix:
1. Prepare csv file to update products (2 products or more). (This csv should contain products with the lowest and the highest id - 1 and 6000)
2. In Admin panel go to: System->Import and choose Product and Add/Update
3. Upload file
4. Check Data
5. Import csv file (error should be displayed now)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22032) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=22032) before we can accept your contribution.<br/><hr/>**Mateusz Wegrzycki** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php [hunk 1]
```diff
@@ -222,7 +222,7 @@ public function getQuery(array $dimensions, string $productType, array $entityId
 
         if ($entityIds !== null) {
             if (count($entityIds) > 1) {
-                $select->where(sprintf('e.entity_id BETWEEN %s AND %s', min($entityIds), max($entityIds)));
+                $select->where('e.entity_id IN(?)', $entityIds);
```
Comment (orlangur) [inline app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php:226]: @kandy @mslabko @duhon do you remember? Something similar was done in other place. Please share your thoughts.

Comment (orlangur) [reply app/code/Magento/Catalog/Model/ResourceModel/Product/Indexer/Price/Query/BaseFinalPrice.php:226]: Yay, I met it accidentally: https://github.com/magento/magento2/pull/19878#discussion_r243217689

**Comment by mtwegrzycki** [comment]
I can see, that related PR #19878 is closed. Can we remove On hold label? There are any requirements to continue this PR?

**Comment by orlangur** [comment]
@mtwegrzycki surely, feel free to proceed with suggested in that PR changes.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

**Review by orlangur — CHANGES_REQUESTED** [review]
@mtwegrzycki surely, feel free to proceed with suggested in that PR changes.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

**Comment by orlangur** [comment]
@mtwegrzycki there must be two `where` conditions after change (see comment I referred to).

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

**Comment by mtwegrzycki** [comment]
Okay, I added new commit with second where conditions.

**Comment by mtwegrzycki** [comment]
I can see, that there any integration errors in Travis. But it does not look like related to my code changes.
Please, let me know, if there are required any additional actions from me.

**Pull request closed without merging**